docs/Roadmap.md

```diff
@@ -0,0 +1,122 @@
+
+The overarching goals of this document are to:
+
+*   Publish areas of current investment.
+*   Encourage external contributors by pointing out feature gaps.
+*   Align this repository with the vision of WebAssembly: a portable technology
+    that is cross-language, cross-platform, and cross-provider.
+
+<a name="abi"></a>
+
+## Spec / ABI
+
+*   (Q1'25: @piotrsikora, @mpwarres) Publish ABI v0.3, containing at least:
+    *   Feature negotiation (proxy-wasm/spec#71 and proxy-wasm/spec#56)
+    *   Better header/body buffering support (proxy-wasm/spec#63)
+    *   Async shared data (proxy-wasm/spec#54)
+    *   Repeated header support (proxy-wasm/spec#53)
+*   (Help wanted) WASI convergence. We want to adopt the component model at WASI
+    1.0. There is a lot of overlap between Proxy-Wasm and some WASI proposals
+    ([wasi-http](https://github.com/WebAssembly/wasi-http),
+    [wasi-keyvalue](https://github.com/WebAssembly/wasi-keyvalue), etc). In the
+    short term, we'd like to define the Proxy-Wasm ABI in
+    [WIT](https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md),
+    to understand:
+    *   How do
+        [Proxy-Wasm interfaces](https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/main/include/proxy-wasm/context_interface.h)
+        map to components?
+    *   What are the API gaps? How should we evolve Proxy-Wasm to become
+        WASI-compatible? What are good incremental steps?
+    *   Are there any performance gaps?
+*   (Help wanted) Evaluate uses of foreign functions to identify feature gaps.
+    *   For example, Envoy
+        [registers foreign functions](https://github.com/search?q=repo%3Aenvoyproxy%2Fenvoy%20RegisterForeignFunction&type=code)
+        for signature checking, compression, filter state, route cache, and CEL
+        expressions.
+    *   Are there similar extensions in Nginx? Apache Traffic Server?
+    *   Which of these features should be promoted to ABI interfaces?
+
+<a name="sdks"></a>
+
+## SDKs / language support
+
+*   (Q1'25: @leonm1) Fork the abandoned Go SDK + support full Golang.
+*   (Google exploring) Build a Python SDK using a MicroPython port.
+*   (Help wanted) Stop using Emscripten in the C++ SDK. Instead use Clang /
+    wasi-sdk (proxy-wasm/proxy-wasm-cpp-sdk#167).
+*   (Help wanted) Merge LeakSignal's
+    [proxy-sdk](https://crates.io/crates/proxy-sdk) crate into the Rust SDK.
+*   (Help wanted) Build a Lua SDK using a Lua interpreter.
+    *   Seems quite feasible given projects like
+        [wasm_lua](https://github.com/vvanders/wasm_lua).
+    *   Could replace Envoy's
+        [Lua filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter)
+    *   Could benefit NGINX's Lua-based [OpenResty](https://openresty.org/)
+        ecosystem
+*   (Help wanted) Optimize Rust SDK binary size. It seems compiler dead-code
+    elimination is thwarted by the use of Context traits.
```

Comment:
Self AI: file issues where missing, especially this one which needs context/details